datafusion/core/src/datasource/physical_plan/parquet.rs

```diff
@@ -430,16 +430,9 @@ struct ParquetOpener {
 impl FileOpener for ParquetOpener {
     fn open(&self, file_meta: FileMeta) -> Result<FileOpenFuture> {
         let file_range = file_meta.range.clone();
-        // When running against the parquet file located at the local file
-        // system, the returned path does not contain a starting `/`, prepend
-        // it here.
-        //
-        // doc: https://docs.rs/object_store/0.7.1/object_store/path/struct.Path.html
-        //
         // NOTE: this is hardcoded for local file system, if we are targeting
         // an object storage, this won't work.
         let mut input_file = file_meta.location().to_string();
```

Comment:
is that mean when data soruce is object store, the file_meta.location() returns nill?

Reply:
My comment here isn't that accurate, it will work under object storage, per the document:

> All paths are relative to the root of the object store

it will still return a path relative to the root of the storage

Reply:
Comment updated